### ISSUE TITLE
pyvips pdfload throwing error - some font thing failed and terminates in the middle with no exception.

### ISSUE THREAD

**Issue by vijtad** [libvips/libvips#1982]
I have a pdf with different width and size of images in different pages.
I am trying to split into different tiff files.
The code terminates in the middle of while loop because of different height with no exception.
However it throws message "some font thing failed" on console.
Can you pls. help me how to resolve it.

```
def splitPDFByPage(pdfFile):
   # Load image
   print("started loading pdf")
   start = timer()
   metadata = pyvips.Image.pdfload(pdfFile, access="sequential")
   mwidth = metadata.get("width")
   mheight = metadata.get("height")
   print(f'Image  {mheight} : {mwidth}' )
   currpage = 0
   n_pages = metadata.get("n-pages")
   print('No of pages - >', n_pages)
   metadata = None
   while(currpage<=n_pages) :
       try:
           output = f'{pdfFile}_{currpage}.tif'
           print(output)
           print(f'Loading pdf {currpage} of {n_pages}' )
           pdf = pyvips.Image.pdfload(pdfFile, page=currpage, fail=False , memory= False, dpi =200,  access="sequential")   
           width = pdf.get("width")
           height = pdf.get("height")
           print(f'Completed loading pdf {currpage} {height} : {width}' )
           print(f'Incremented loading pdf {currpage} of {n_pages}' )
           pdf[0].tiffsave(output, resunit="inch", squash = True, xres =200,yres=200, tile=False, compression="ccittfax4", bitdepth= 1, level=1 )
           print(f'Saved loading pdf {currpage} of {n_pages}' )
           currpage = currpage + 1
           pdf = None
       except Exception as e:
            print(str(e))
       finally:
            # no effect
            gc.collect()


   output = pdfFile +'_Merged.tif'
   print('output',output)
   with TiffImagePlugin.AppendingTiffWriter(output,True) as tf:        
       for tiff_in in range(0,n_pages):
           input = f'{pdfFile}_{tiff_in}.tif'
           im = Image.open(input,"r")
           im.save(tf)
           tf.newFrame()
           print('in')
           im.close()   

   pdf = None
   end = timer()
   print("Time in seconds", end - start)
   return 

```

**Comment by jcupitt** [comment]
Hello @vijtad,

Try changing:

```python
           pdf[0].tiffsave(output, resunit="inch", squash = True, xres =200,yres=200, tile=False, compression="ccittfax4", bitdepth= 1, level=1 )
```

To:

```python
           pdf.tiffsave(output, resunit="inch", squash = True, xres =200,yres=200, tile=False, compression="ccittfax4", bitdepth= 1, level=1 )
```

You can change:

```python
   mwidth = metadata.get("width")
```

To:

```python
   mwidth = metadata.width
```

As well, I expect you know.

**Comment by vijtad** [comment]
For my test, I am using docker build on pyvips-python3.7-alpine docker. It crashed because I assigned less  memory.
My sample test PDF has 2273 pages with images of different height and width.

I would like to compress the image as ccittfax4 and create multi-page TIFF file which has different dimensions in PDF.
Pls suggest me how can I convert PDF to TIFF with different dimensions merged into one TIFF as multi-tiff pages.

I get error when I changed the code to -

``
pdf.tiffsave(output, resunit="inch", squash = True, xres =200,yres=200, tile=False, compression="ccittfax4", strip=True, bitdepth= 1, level=1,bigtiff=True )
``

Console log is 
 
  **Loading pdf 1 of 2273
Completed loading pdf 1 2199 : 1700
Incremented loading pdf 1 of 2273
some font thing failed
some font thing failed
some font thing failed
some font thing failed
some font thing failed
some font thing failed
some font thing failed
some font thing failed
some font thing failed
some font thing failed
some font thing failed
some font thing failed
some font thing failed
some font thing failed
unable to call tiffsave
  Fax3SetupState: Bits/sample must be 1 for Group 3/4 encoding/decoding
wbuffer_write: write failed
unix error: No error information**

Using **pdf[0]** fixed this error.
